jira-config/jira-config-platform/src/main/java/org/xwiki/contrib/jira/config/internal/JIRAConfigClassDocumentConfigurationSource.java

```diff
@@ -107,13 +120,19 @@ protected Object getBaseProperty(String propertyName, boolean text) throws XWiki
             StringProperty idProperty = (StringProperty) baseObject.getField("id");
             StringProperty urlProperty = (StringProperty) baseObject.getField("url");
             if (!isPropertyEmpty(idProperty) && !isPropertyEmpty(urlProperty)) {
-                StringProperty usernameProperty = (StringProperty) baseObject.getField("username");
-                StringProperty passswordProperty = (StringProperty) baseObject.getField("password");
-                if (!isPropertyEmpty(usernameProperty) && !isPropertyEmpty(passswordProperty)) {
-                    jiraServer = new JIRAServer(urlProperty.getValue(),
-                        usernameProperty.getValue(), passswordProperty.getValue());
+                String authType = baseObject.getStringValue("authenticationType");
+                if (StringUtils.isEmpty(authType) || "noAuth".equals(authType)) {
+                    jiraServer = new JIRAServer(urlProperty.getValue(), idProperty.getValue());
                 } else {
-                    jiraServer = new JIRAServer(urlProperty.getValue());
+                    try {
+                        JIRAuthenticatorFactory jirAuthenticatorFactory = componentManagerProvider.get()
+                            .getInstance(JIRAuthenticatorFactory.class, authType);
+                        jiraServer = new JIRAServer(urlProperty.getValue(), idProperty.getValue(),
+                            jirAuthenticatorFactory.get(idProperty.getValue()));
+                    } catch (ComponentLookupException e) {
+                        logger.error("Can't find JIRAuthenticatorFactory with name [{}]", authType, e);
```

Comment:
I think I'd do it differently: here it's easy for an admin to entirely miss the fact that a configured server instance doesn't work because of a bad authenticator name without having any clue in the UI. Also you're mixing the fact that the component hint isn't right and the fact that the component cannot be properly initialized. 
So I'd use a if to check if the hint exists first, if it's not the case I'd call the JIRAServer constructor with null, and add some checks in JIRAServer to test if it's null and be able to display that in the config UI. And then if the hint exists but there's a problem to instantiate it I'd throw an exception.